packages/workbox-webpack-plugin/src/lib/read-file-wrapper.js

```diff
@@ -28,7 +28,7 @@
  */
 function readFileWrapper(readFileFn, filePath) {
   return new Promise((resolve, reject) => {
-    readFileFn(filePath, 'utf8', (error, data) => {
```

Comment:
Would the `utf8` encoding option cause any problem? This seems like the only different in calling the readFile API.

Reply:
Yes - it causes a runtime error, since the method from the public API only expects two parameters, the file path and the callback. See the API of the `CachedInputFileSystem` for an example: https://github.com/webpack/enhanced-resolve/blob/master/lib/CachedInputFileSystem.js#L238

Reply:
A few data points:

- Two-parameters seems to match how `webpack` calls that function, as per https://github.com/webpack/webpack/blob/eca7bad8de54c39b9cb8b138793362b8a17ac11b/lib/Compiler.js#L349

- `memory-fs` has support for both two-parameter and three-parameter modes: https://github.com/webpack/memory-fs/blob/7ae73ebf34ae2f95266e6c3fef1f8e99f9d2308a/lib/MemoryFileSystem.js#L313

The one usage of `readFileWrapper` is to read in the `swSrc`, and the results of that are then used as part of a ES2015 template string to construct the final `swDest` file.

If the native `fs.readFile()` is called in two-parameter mode, without specifying the encoding, then it returns the contents as a `Buffer` rather than a `string`. *But*, using the `Buffer` within the final ES2015 template string will automatically call `toString()` on in, and the default encoding for that stringification is `utf-8`. So, we should have the same end result for the `fs.readFile()` case as well.

Reply:
👍 Awesome, thank you for the detailed explanation and references to the source code.